generator/model/events.py

```diff
@@ -1,55 +1,53 @@
-from random import random
-from typing import List, Dict
+from typing import List, Dict, Set
 from generator.config.basic import triggers
 from generator.model.event import Event
 
 
 class Events:
     def __init__(self):
         self.events: Dict[str, List[any]] = self._get_all_possible_events()
-        self.current_events: List[Event] = []
+        self.current_events: Set[Event] = set()
 
     def __hash__(self):
         return hash(self.events)
 
     def block_event(self, event: Event):
-        self.current_events.append(event)
+        self.current_events.add(event)
 
     def release_event(self, event: Event):
         self.current_events.remove(event)
 
     def update_blocked(self, current_time: int):
-        for _ in self.current_events:
-            if _.time < current_time:
-                self.release_event(_)
+        to_remove = []
+        for e in self.current_events:
+            if e.time < current_time:
+                to_remove.append(e)
 
-    def get_events_keys(self) -> List[str]:
-        return list(self.events.keys())
+        for e in to_remove:
+            self.release_event(e)
+
+    @staticmethod
+    def get_rooms() -> List[str]:
+        return list(triggers.keys())
 
     @staticmethod
     def _get_all_possible_events() -> Dict[str, List[any]]:
         all_possible: Dict[str, List[any]] = {}
         for room in triggers:
+            all_possible[room] = []
             for trigger in triggers[room]:
-                all_possible[trigger] = []
                 for event in triggers[room][trigger]:
                     if type(event) is tuple:
-                        all_possible[trigger].append(((trigger, event[0]), (trigger, event[1])))
+                        all_possible[room].append(((trigger, event[0]), (trigger, event[1])))
                     else:
-                        all_possible[trigger].append((trigger, event))
+                        all_possible[room].append((trigger, event))
         return all_possible
 
-    def generate(self, room: str, current_time: int) -> Event:
+    def generate(self, room: str, current_time: int) -> any:
         self.update_blocked(current_time)
         events_pool = self.events[room]
-        while True:
-            is_free: bool = True
-            next_event: Event = events_pool[int(random() * len(events_pool))]
+        for next_event in events_pool:
             if type(next_event) is tuple:
-                for _ in self.current_events:
-                    if next_event[0][0] == _.trigger and next_event[0][1] == _.name:
-                        is_free = False
-                        break
-            if is_free:
-                break
-        return next_event
+                if next_event[1] not in set(map(lambda e: (e.trigger, e.name), self.current_events)):
```

Comment:
Można zrobić tak ` if next_event[1] not in map(lambda e: (e.trigger, e.name), self.current_events):`